packages/react-components/react-menu/library/src/components/Menu/useMenu.tsx

```diff
@@ -63,6 +63,7 @@ export const useMenu_unstable = (props: MenuProps): MenuState => {
     align: isSubmenu ? 'top' : 'start',
     target: props.openOnContext ? contextTarget : undefined,
     fallbackPositions: isSubmenu ? submenuFallbackPositions : undefined,
+    autoSize: true,
```

Comment:
If we decide to go this path, we also should have a plan on how to enable it for other components (`Popover`, `Combobox`, etc.). Probably, enabling it in `usePositioning()` as a default would be a nice option.